staging/src/k8s.io/apimachinery/pkg/util/managedfields/internal/structuredmerge.go

```diff
@@ -44,33 +44,38 @@ func NewStructuredMergeManager(typeConverter TypeConverter, objectConverter runt
 	if typeConverter == nil {
 		return nil, fmt.Errorf("typeconverter must not be nil")
 	}
+	builder := merge.UpdaterBuilder{
+		// This is the converter provided to SMD from k8s
+		Converter:         newVersionConverter(typeConverter, objectConverter, hub),
+		IgnoredFields:     resetFields,
+		ReturnInputOnNoop: true,
+	}
 	return &structuredMergeManager{
 		typeConverter:   typeConverter,
 		objectConverter: objectConverter,
 		objectDefaulter: objectDefaulter,
 		groupVersion:    gv,
 		hubVersion:      hub,
-		updater: merge.Updater{
-			Converter:     newVersionConverter(typeConverter, objectConverter, hub), // This is the converter provided to SMD from k8s
-			IgnoredFields: resetFields,
-		},
+		updater:         builder.BuildUpdater(),
 	}, nil
 }
 
 // NewCRDStructuredMergeManager creates a new Manager specifically for
 // CRDs. This allows for the possibility of fields which are not defined
 // in models, as well as having no models defined at all.
 func NewCRDStructuredMergeManager(typeConverter TypeConverter, objectConverter runtime.ObjectConvertor, objectDefaulter runtime.ObjectDefaulter, gv schema.GroupVersion, hub schema.GroupVersion, resetFields map[fieldpath.APIVersion]*fieldpath.Set) (_ Manager, err error) {
+	builder := merge.UpdaterBuilder{
+		Converter:         newCRDVersionConverter(typeConverter, objectConverter, hub),
+		IgnoredFields:     resetFields,
+		ReturnInputOnNoop: true,
```

Comment:
Is it correct that we are now relying on the `IgnoreManagedFieldsTimestampsTransformer` to avoid changing timestamp? Or is the new comparison code in SMD being called somewhere?
 https://github.com/kubernetes/kubernetes/blob/c5a8015bbae1e1095813d8e0719f017477dc7112/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/equality.go#L75

Reply:
The former, as far as I can tell, the `IgnoreManagedFieldsTimestampsTransformer` is making this useless anyway. I would expect the integration tests there to be redundant with these tests.

Reply:
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/120703/pull-kubernetes-unit/1702714958216695808

hmm looks like its not working 0_o

Reply:
ah, that test does not run the transformer

Reply:
> I would expect the integration tests there to be redundant with these tests.

I can't find any that specifically test the cases that motivated this PR, or any that check managed fields timestamp is not updated.